### PR TITLE
Fix PHP8 compatibility issues

### PR DESCRIPTION
**Pull request by Pebblo**
This PR just prevents fatal's from PHP8

Swaps out instances of each() for foreach()
Adjusts the parameter order passed to implode()
Updates FPDF to the latest version so its works with PHP8.
Fixes a couple of notices.
Patches the Stripe SDK with a couple of fixes.
Uses the 'new' style constructor within PayPal Pro

### CODE REVIEW

**Comment by garthkoyle** [comment]
Thank you.

File: gateways/paypal_pro/includes/paypal.nvp.class.php [hunk 1]
```diff
@@ -72,6 +73,7 @@ function Espresso_PayPal($DataArray)
 		$this -> NVPCredentials .= $this -> APISubject != '' ? 'SUBJECT=' . $this -> APISubject : '';
 		$this -> NVPCredentials .= $this -> APIMode == 'Signature' ? '&SIGNATURE=' . $this -> APISignature : '';
 
+		var_dump($this->NVPCredentials);
```
Comment (tn3rb) [inline gateways/paypal_pro/includes/paypal.nvp.class.php]: OOPS!

going to go out on a limb here and say it's maybe not a good thing to dump credentials like this 🤔

File: gateways/stripe/do_transaction.php [hunk 2]
```diff
@@ -38,7 +38,11 @@ function espresso_process_stripe($payment_data) {
 			$payment_intent_id
 		);
 		//Convert the Stripe payment intent to an array so we can use it.
-		$intent_array = $intent->__toArray(true);
+		if(method_exists($intent, 'toArray')) {
+			$intent_array = $intent->toArray(true);
+		} else {
+			$intent_array = $intent->__toArray(true);
+		}
```
Comment (tn3rb) [inline gateways/stripe/do_transaction.php]: not critical, but it's preferable to use ternaries:

```php
$intent_array = method_exists($intent, 'toArray')
    ? $intent->toArray(true)
    : $intent->__toArray(true);
```

File: includes/admin-reports/enter_attendee_payments.php [hunk 3]
```diff
@@ -17,10 +17,11 @@ function enter_attendee_payments() {
 </div>';
 
 
-
+	//var_dump($_POST);
+	//var_dump($_REQUEST);
```
Comment (tn3rb) [inline includes/admin-reports/enter_attendee_payments.php]: maybe remove?

File: includes/admin-reports/enter_attendee_payments.php [hunk 4]
```diff
@@ -17,10 +17,11 @@ function enter_attendee_payments() {
...
 	$multi_reg = FALSE;
-	$event_id = isset($_POST['event_id']) ? absint( $_POST['event_id'] ) : isset($_REQUEST['event_id']) ? absint( $_REQUEST['event_id'] ) : '';
-	$registration_id = isset( $_POST['registration_id'] ) ? sanitize_text_field( $_POST['registration_id'] ) : isset( $_REQUEST['registration_id'] ) ? sanitize_text_field( $_REQUEST['registration_id'] ) : FALSE;
+	$event_id = isset($_REQUEST['event_id']) ? absint( $_REQUEST['event_id'] ) : '';
```
Comment (tn3rb) [inline includes/admin-reports/enter_attendee_payments.php:20]: LOLZ! MUCH better 💯

File: includes/event-management/update_event.php [hunk 5]
```diff
@@ -40,8 +40,8 @@ function update_event($recurrence_arr = array()) {
                 $re_manual_end_dates = !empty($_POST['recurrence_manual_end_dates']) ? (array)$_POST['recurrence_manual_end_dates'] : array();
                 $re_manual_end_dates = array_map('sanitize_text_field', $re_manual_end_dates);
                 $re_params = array(
-                    'start_date'					=> !empty($_POST['recurrence_start_date']) ? sanitize_text_field($_POST['recurrence_start_date']) :'',
-                    'event_end_date'				=> !empty($_POST['recurrence_event_end_date']) ? sanitize_text_field($_POST['recurrence_event_end_date']) : '',
+                    'start_date'					=> !empty($_POST['recurrence_start_date']) ? sanitize_text_field($_POST['recurrence_start_date']) : '0000-00-00',
+                    'event_end_date'				=> !empty($_POST['recurrence_event_end_date']) ? sanitize_text_field($_POST['recurrence_event_end_date']) : '0000-00-00',
```
Comment (tn3rb) [inline includes/event-management/update_event.php:44]: 👍🏻

File: gateways/stripe/stripe-php-6.43.1/lib/ApiRequestor.php [hunk 6]
```diff
@@ -399,7 +399,7 @@ private function _requestRaw($method, $url, $params, $headers)
             $hasFile
         );
 
-        if (array_key_exists('request-id', $rheaders)) {
+        if (isset($rheaders['request-id'])) {
```
Comment (tn3rb) [inline gateways/stripe/stripe-php-6.43.1/lib/ApiRequestor.php:402]: subtle but impactful change 👍🏻

**Review by tn3rb — DISMISSED** [review]
Found a `var_dump()` of potentially sensitive data that needs to be removed, and had a few other minor suggestions, but overall good stuff

**Pull request merged**